fabric-common/types/index.d.ts

```diff
@@ -226,10 +230,14 @@ export class DiscoveryService extends ServiceAction {
 	public getDiscoveryResults(refresh?: boolean): Promise<any>;
 	public close(): void;
 }
+export interface RegistrationOpts {
+	startBlock?: number|string|Long;
+	endBlock?: number|string|Long;
+	unregister?: boolean;
+}
 
-export class EventListener {
-	constructor(listenerType: string, callback: any, options: any, event: string);
-	onEvent(error: Error, event: any): void;
+export interface EventListener {
+	onEvent(error: Error, event: EventInfo): void;
```

Comment:
This should not be in the public interface

Reply:
Agree, WIP